src/AbstractApplication.php

```diff
@@ -131,13 +159,54 @@ public function get($key, $default = null)
 	 */
 	public function getLogger()
 	{
-		// If a logger hasn't been set, use NullLogger
-		if (! ($this->logger instanceof LoggerInterface))
+		return $this->getContainer()->get('Psr\\Log\\LoggerInterface');
```

Comment:
This feels like a step backwards to me.  Right now our Framework classes that support a PSR-3 logger expect them to be injected through the LoggerAwareInterface `setLogger` method as dependencies.  This change (and others in this PR) basically turn our DIC into a service locator and breaks some of that concept of dependency injection.